app/src/main/java/com/seoulfitu/android/ui/facility/SportsFacilityViewModel.kt

```diff
@@ -44,31 +45,35 @@ constructor(
     private val _detailOpenEvent: MutableLiveData<UiSportsFacility> = MutableLiveData()
     val detailOpenEvent: LiveData<UiSportsFacility> = _detailOpenEvent
 
+    private val _selectedOptions: MutableLiveData<UiSelectedOptions?> = MutableLiveData()
+    val selectedOptions: LiveData<UiSelectedOptions?> = _selectedOptions
+
     fun getAllFacilities() {
         viewModelScope.launch {
             repository.getSportsFacility().onSuccess { facilities ->
                 _sportsFacilities.value =
                     facilities.map { facility ->
-                        val isFacilityScraped = isScraped(facility)
+                        val isFacilityScraped = isScraped(facility.info.facilityName)
                         facility.toUi(isFacilityScraped)
                     }
                 _listSportsFacilities.value =
                     UiSportsFacilityList(_sportsFacilities.value ?: emptyList())
 
-                _sportsFacilities.value?.forEach { searchPosition(it) }
-//                _facilityWithCoordinate.value = null
+                _sportsFacilities.value?.forEachIndexed { index, uiSportsFacility ->
+                    searchPosition(index, uiSportsFacility)
+                }
             }
         }
     }
 
-    private suspend fun isScraped(sportsFacility: SportsFacility): Boolean {
+    private suspend fun isScraped(facilityName: String): Boolean {
         return withContext(Dispatchers.IO) {
             val scrapedFacilities = scrapRepository.getAll()
-            scrapedFacilities.any { it.info.facilityName == sportsFacility.info.facilityName }
+            scrapedFacilities.any { it.info.facilityName == facilityName }
         }
     }
 
-    private suspend fun searchPosition(sportsFacility: UiSportsFacility) {
+    private suspend fun searchPosition(facilityIndex: Int, sportsFacility: UiSportsFacility) {
```

Comment:
```suggestion
    private suspend fun searchPosition() {
        viewModelScope.launch {
            _sportsFacilities.value?.map { sportsFacility ->
                async { sportsFacility to geocodingRepository.geocode(sportsFacility.address) }
            }?.awaitAll()?.forEach { (sportsFacility, addresses) ->
                addresses.onSuccess { address ->
                    if (address.values.isNotEmpty()) {
                        val cor = address.values.first().coordinate
                        _facilityWithCoordinate.value =
                            UiSportsFacilityWithCoordinate(sportsFacility, cor.x, cor.y)
                    }
                }
            }
            _facilityWithCoordinate.value = null
        }
    }
```
이렇게 `async`랑 `awaitAll()`써서 일이 다 끝날때까지 기다렸다가 마커 다 박아주고 Null로 바꿔서 마커 다 찍힌 후에 로딩 창 없애기 가능할 것 같아요! 
Index를 활용하는 방법도 좋습니당😘

Reply:
마커들이 팝팝팝 뜨냐, 한번에 쫜 나타나냐의 차이인 것 같은데
async를 사용하는게 뭔가 간지폭풍이어서 저는 뽀또의 피드백대로 하겠습니다!😉